### PR TITLE
PLAT-1036 | Content classifier in admin

### PR DESCRIPTION
**Pull request by rogatty**
https://bethink.atlassian.net/browse/PLAT-1036

### CODE REVIEW

File: resources/assets/js/admin/components/flashcards/edit/FlashcardEditor.vue [hunk 1]
```diff
@@ -126,12 +145,21 @@ export default {
 	},
 	async mounted() {
 		if (this.isEdit) {
+			await this.setupCurrentUser();
```
Comment (kvas-damian) [inline resources/assets/js/admin/components/flashcards/edit/FlashcardEditor.vue]: Why do we need this?

Comment (rogatty) [reply resources/assets/js/admin/components/flashcards/edit/FlashcardEditor.vue]: `fetchTaxonomyTerms` uses `getters.canAccess` which is based on the current user. In some places the user is already loaded, in some (like here) it's not, so we need to wait.

Comment (rogatty) [reply resources/assets/js/admin/components/flashcards/edit/FlashcardEditor.vue]: @kvas-damian @bognix @gitkuba what do you think about making all getters (apart from `isCurrentUserLoading`) in the `currentUser` async and blocked by `setupCurrentUser`?
https://github.com/bethinkpl/wnl-platform/blob/31099ab106e39d5ee61bc1304f7ea1368fa28584/resources/assets/js/store/modules/currentUser.js#L39-L61
IMO it makes sense and would prevent many race conditions but I'm a bit worried about possible side effects.

Comment (rogatty) [reply resources/assets/js/admin/components/flashcards/edit/FlashcardEditor.vue]: Let's follow up in https://bethink.atlassian.net/browse/PLAT-1060

File: resources/assets/js/admin/components/flashcards/edit/FlashcardEditor.vue [hunk 2]
```diff
@@ -126,12 +145,21 @@ export default {
...
+
 			const response = await this.form.populate(this.flashcardResourceUrl, ['include']);
 			const tags = response.tags;
 			if (!tags) return;
 
 			this.flashcardTags = tags.map(tagId => response.included.tags[tagId]);
 		}
+	},
+	watch: {
+		async 'flashcardId'() {
```
Comment (bognix) [inline resources/assets/js/admin/components/flashcards/edit/FlashcardEditor.vue]: you can skip the `' '`

File: resources/assets/js/admin/components/quizes/edit/QuizQuestionEditor.vue [hunk 3]
```diff
@@ -330,5 +357,11 @@ export default {
 			this.setupFreshQuestion();
 		}
 	},
+	async mounted() {
+		if (this.isEdit) {
+			await this.setupCurrentUser();
+			await this.fetchTaxonomyTerms({contentType: CONTENT_TYPES.QUIZ_QUESTION, contentIds: [this.$route.params.quizId]});
```
Comment (bognix) [inline resources/assets/js/admin/components/quizes/edit/QuizQuestionEditor.vue]: I would consider putting `this.$route.params.quizId` to compute or data

File: resources/assets/js/admin/components/slides/EditSlide.vue [hunk 4]
```diff
@@ -46,12 +64,12 @@ export default {
...
 			this.slideId = event.target.value;
 		},
-		onResourceUrlFetched({url, slideId}) {
+		async onResourceUrlFetched({url, slideId}) {
 			this.slideId = slideId;
 			this.resourceUrl = url;
 		}
 	},
-	mounted() {
+	async mounted() {
```
Comment (bognix) [inline resources/assets/js/admin/components/slides/EditSlide.vue]: I think async is redundant here.

File: resources/assets/js/admin/components/slides/EditSlide.vue [hunk 5]
```diff
@@ -46,12 +64,12 @@ export default {
 		saveSlideId(event) {
 			this.slideId = event.target.value;
 		},
-		onResourceUrlFetched({url, slideId}) {
+		async onResourceUrlFetched({url, slideId}) {
```
Comment (bognix) [inline resources/assets/js/admin/components/slides/EditSlide.vue]: I think async is redundant here.

File: resources/assets/js/admin/components/flashcards/edit/FlashcardEditor.vue (hunk 1 above)
Comment (kvas-damian) [reply resources/assets/js/admin/components/flashcards/edit/FlashcardEditor.vue]: Maybe we could perform `await` in `fetchTaxonomyTerms` method?
Although it doesn't solve all use-cases.

File: resources/assets/js/admin/components/quizes/edit/QuizQuestionEdit.vue [hunk 6]
```diff
@@ -1,14 +1,13 @@
 <template>
-	<wnl-quiz-question-editor :is-edit="true"></wnl-quiz-question-editor>
+	<wnl-quiz-question-editor :quiz-question-id="$route.params.quizQuestionId" />
```
Comment (kvas-damian) [inline resources/assets/js/admin/components/quizes/edit/QuizQuestionEdit.vue]: Can we pass it as props from the router?

Comment (rogatty) [reply resources/assets/js/admin/components/quizes/edit/QuizQuestionEdit.vue]: 👍 https://github.com/bethinkpl/wnl-platform/pull/1323/commits/7dc22daa15ae0d6328430b41990522158e5a0ac7

File: resources/assets/js/admin/components/quizes/edit/QuizQuestionEditor.vue [hunk 7]
```diff
@@ -319,16 +348,23 @@ export default {
 		questionExplanation(val) {
 			if (val) this.$refs.explanationEditor.editor.innerHTML = val;
 		},
-		'$route.params.quizId'(quizId) {
-			this.getQuizQuestion(this.$route.params.quizId);
+		quizQuestionId(quizQuestionId) {
+			this.getQuizQuestion(quizQuestionId);
+			this.fetchTaxonomyTerms({contentType: CONTENT_TYPES.QUIZ_QUESTION, contentIds: [quizQuestionId]});
```
Comment (kvas-damian) [inline resources/assets/js/admin/components/quizes/edit/QuizQuestionEditor.vue]: Do we need this fetch? We already fetch terms in `mounted` hook. Maybe we don't need this watch at all?

Comment (rogatty) [reply resources/assets/js/admin/components/quizes/edit/QuizQuestionEditor.vue]: It's used when user creates new quiz question.

**Pull request merged**